crypto/src/main/scala/org/bitcoins/crypto/DLEQUtil.scala

```diff
@@ -15,26 +14,32 @@ object DLEQUtil {
   }
 
   def dleqNonceFunc(
-      hash: ByteVector,
       fe: FieldElement,
-      algoName: String): FieldElement = {
-    val kBytes =
-      CryptoUtil.taggedSha256(fe.bytes ++ hash, algoName).bytes
-    FieldElement(kBytes)
+      adaptorPoint: ECPublicKey,
+      point: ECPublicKey,
+      tweakedPoint: ECPublicKey,
+      auxRand: ByteVector): FieldElement = {
+    val hash = CryptoUtil
+      .sha256(point.compressed.bytes ++ tweakedPoint.compressed.bytes)
+      .bytes
+
+    AdaptorUtil.adaptorNonce(hash,
+                             fe.toPrivateKey,
+                             adaptorPoint,
+                             "DLEQ",
+                             auxRand)
   }
 
   def dleqChallengeHash(
```

Comment:
I think this is the best reference for a scaladoc? 

https://github.com/discreetlogcontracts/dlcspecs/blob/d01595b70269d4204b05510d19bba6a4f4fcff23/ECDSA-adaptor.md#proof-of-discrete-logarithm-equality